src/coll_score/ucc_coll_score.c

```diff
@@ -533,46 +680,81 @@ static ucc_status_t ucc_coll_score_update_one(ucc_list_link_t *dest,
     while (s != src && d != dest) {
         rs = ucc_container_of(s, ucc_msg_range_t, list_elem);
         rd = ucc_container_of(d, ucc_msg_range_t, list_elem);
+        ucc_assert((NULL == rs->init) || (NULL != rs->team));
         if (rd->start >= rs->end) {
             /* skip src range - no overlap */
-            s = s->next;
+            s   = s->next;
+            if (rs->init) {
```

Comment:
in what case rs->init is NULL? Seems like in ucc_coll_score_parse_str we set it to alg_init or general init